jwst/refpix/irs2_subtract_reference.py

```diff
@@ -138,6 +138,30 @@ def correct_model(input_model, irs2_model,
     # or interspersed reference pixels).
     irs2_mask = make_irs2_mask(output_model, scipix_n, refpix_r)
 
+    # If the IRS2 reference file includes data quality info, use that to
+    # set bad reference pixel values to zero.
+    if hasattr(irs2_model, 'dq_table') and len(irs2_model.dq_table) > 0:
+        output = irs2_model.dq_table.field("output")
+        odd_even = irs2_model.dq_table.field("odd_even")
+        mask = irs2_model.dq_table.field("mask")
+        got_dq = True
+    else:
+        log.warning("DQ extension not found in reference file")
+        got_dq = False
+
+    if got_dq:
+        for integ in range(n_int):
+            for row in range(len(output)):
+                part = output[row]
+                bits = decode_mask(output, mask, row)
+                log.debug("output {} DQ bits {}".format(part, bits))
+                for k in bits:
+                    ref = (scipix_n // 2 + k * (scipix_n + refpix_r) +
+                           2 * (odd_even[row] - 1))
+                    log.debug("bad interleaved reference at pixels {} {}"
+                              .format(ref, ref+1))
+                    data[integ, :, :, ref:ref+2] = 0.
```

Comment:
Could leave some comment here as to what is happening?  Why dividing, multiplying and adding 2?